src/generator/protocol/operations.ts

```diff
@@ -83,6 +83,70 @@ class methodSig implements MethodSig {
   }
 }
 
+function formatQueryParamValue(param: Parameter, imports: ImportManager): string {
+  let separator = ',';
+  switch (param.protocol.http?.style) {
+    case SerializationStyle.PipeDelimited:
+      separator = '|';
+      break;
+    case SerializationStyle.SpaceDelimited:
+      separator = ' ';
+      break;
+    case SerializationStyle.TabDelimited:
+      separator = '\\t';
+      break;
+  }
+  switch (param.schema.type) {
+    case SchemaType.Array:
+      const arraySchema = <ArraySchema>param.schema;
+      switch (arraySchema.elementType.type) {
+        case SchemaType.Choice:
+        case SchemaType.SealedChoice:
+        case SchemaType.String:
+          imports.add('strings');
+          return `strings.Join(${param.language.go!.name}, "${separator}")`;
+        default:
+          imports.add('fmt');
+          imports.add('strings');
+          return `strings.Join(strings.Fields(strings.Trim(fmt.Sprint(${param.language.go!.name}), "[]")), "${separator}")`;
+      }
+    case SchemaType.Boolean:
+      imports.add('strconv');
+      return `strconv.FormatBool(${param.language.go!.name})`;
+    case SchemaType.ByteArray:
+      // ByteArray is a base-64 encoded value in string format
+      return `string(${param.language.go!.name})`;
+    case SchemaType.Choice:
+    case SchemaType.SealedChoice:
+      return `string(${param.language.go!.name})`;
+    case SchemaType.Constant:
+      const constSchema = <ConstantSchema>param.schema;
+      // cannot use formatConstantValue() since all values are treated as strings
+      return `"${constSchema.value.value}"`;
+    case SchemaType.Date:
+    case SchemaType.DateTime:
+      return `${param.language.go!.name}.String()`;
+    case SchemaType.Integer:
+      imports.add('strconv');
+      const intSchema = <NumberSchema>param.schema;
+      let intParam = param.language.go!.name;
+      if (intSchema.precision === 32) {
+        intParam = `int64(${intParam})`;
```

Comment:
Just checking, why is this int64 and not int32?

Reply:
`strconv.FormatInt()` takes an `int64` as its first parameter so we have to up-cast in this case (same is true for `FormatFloat` below).

Reply:
Ah I see, thanks for the explanation!